python-spec/src/somacore/data.py

```diff
@@ -170,6 +185,14 @@ def index_column_names(self) -> Tuple[str, ...]:
         """
         raise NotImplementedError()
 
+    @property
+    def domains(self) -> Tuple[Tuple[Any, Any], ...]:
```

Comment:
I'm now thinking this should be singular `domain` in line with tiledb usage.

If everyone's agreed I'll of course make the same change in the impl PR https://github.com/single-cell-data/TileDB-SOMA/pull/1027.

Reply:
I prefer the singular as well.  @thetorpedodog ?